dotCMS/src/main/java/com/dotmarketing/util/ImportUtil.java

```diff
@@ -410,32 +428,53 @@ public static ImportResult importFileResult(
+
+                                // Handle transaction commits based on successful imports instead of line number
+                                if (successfulImports % params.commitGranularityOverride() == 0) {
+                                    handleBatchCommit(params.preview(), lineNumber);
+                                    counters.incCommits();
+                                    savepoint = null;
+
+                                    Logger.debug(ImportUtil.class,
+                                            "Committed batch at line " + lineNumber + " after " +
+                                                    successfulImports + " successful imports.");
+                                }
                             } else {
                                 messages.add(ValidationMessage.builder()
                                         .type(ValidationMessageType.ERROR)
                                         .code(LANGUAGE_NOT_FOUND.name())
-                                        .message(LanguageUtil.get(user,
+                                        .message(LanguageUtil.get(params.user(),
                                                 "Locale-not-found-for-languageCode") + " ='"
-                                                + csvLine[languageCodeHeaderColumn]
+                                                + csvLine[params.languageCodeHeaderColumn()]
                                                 + "' countryCode='"
-                                                + csvLine[countryCodeHeaderColumn] + "'")
+                                                + csvLine[params.countryCodeHeaderColumn()] + "'")
                                         .context(Map.of(
-                                                "Language code", csvLine[languageCodeHeaderColumn],
-                                                "Country code", csvLine[countryCodeHeaderColumn]
+                                                "Language code", csvLine[params.languageCodeHeaderColumn()],
+                                                "Country code", csvLine[params.countryCodeHeaderColumn()]
                                         ))
                                         .lineNumber(lineNumber)
                                         .build());
-                            }
 
-                            // Handle transaction commits
-                            if (lineNumber % commitGranularity == 0) {
-                                handleBatchCommit(preview, lineNumber);
+                                // Count language errors as failures
+                                failedRows++;
                             }
 
                         } catch (Exception ex) {
 
                             failedRows++;
 
+                            // Roll back to savepoint on error
+                            if (!params.preview() && savepoint != null && !params.stopOnError()) {
```

Comment:
I think you may be missing the rollback when the savepoint is null

Reply:
We're now using a savePoint at all times, it shouldn't ever be null. This null check is just a precaution. Since it gets it gets cleaned up 
But it should never be null